src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -95,6 +95,12 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
             return null;
         }
 
+        /// <summary>
+        /// Retrieves all known messages <see cref="MessageMetadata" />.
+        /// </summary>
+        /// <returns>An array of <see cref="MessageMetadata" /> for all known message.</returns>
+        public MessageMetadata[] GetAllMessages() => messages.Values.ToArray();
```

Comment:
Should we add a remarks that this method materializes the array every time it is called and the caller is responsible for caching the result?